tests/schema/tests.py

```diff
@@ -2359,3 +2359,29 @@ class Meta:
         doc = Document.objects.create(name='Test Name')
         student = Student.objects.create(name='Some man')
         doc.students.add(student)
+
+    def test_rename_table_renames_deferred_sql_references(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(Book)
+            editor.alter_db_table(Author, 'schema_author', 'schema_renamed_author')
+            editor.alter_db_table(Author, 'schema_book', 'schema_renamed_book')
+            for statement in editor.deferred_sql:
```

Comment:
Something like `self.assertGreater(len(editor.deferred_sql), 0)` might be appropriate to ensure there is `deferred_sql`. I'm always cautious about assertions within loops without something to check that the loop is executed.